### PR TITLE
Update quaternions related operations

### PR DESCRIPTION
**Pull request by jmirabel**
- fix uniform sampling of quaternions
- update doc
- update assert threshold

I think it would be nice to add a unittest doing plenty of integration for a spherical joint and checking that the norm of the quaternion is always close to 1.

### CODE REVIEW

**Comment by jmirabel** [comment]
Related to #328

File: src/math/quaternion.hpp [hunk 1]
```diff
@@ -81,5 +81,28 @@ namespace se3
     const_cast <Eigen::QuaternionBase<D> &> (q).coeffs() *= alpha;
   }
 
+  /// Uniformly random quaternion sphere.
+  template <typename D>
+  void uniformRandom (const Eigen::QuaternionBase<D> & q)
+  {
+    // EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(D, 4);
```
Comment (jcarpent) [inline src/math/quaternion.hpp]: You can remove this line.

File: src/math/quaternion.hpp [hunk 2]
```diff
@@ -81,5 +81,28 @@ namespace se3
...
+    const Scalar u2 = (Scalar)rand() / RAND_MAX;
+    const Scalar u3 = (Scalar)rand() / RAND_MAX;
+
+    const Scalar mult1 = sqrt (1-u1);
+    const Scalar mult2 = sqrt (u1);
+
+    Scalar s2,c2; SINCOS(2.*PI*u2,&s2,&c2);
+    Scalar s3,c3; SINCOS(2.*PI*u3,&s3,&c3);
+
+    const_cast <Eigen::QuaternionBase<D> &> (q).w() = mult1 * s2;
```
Comment (jcarpent) [inline src/math/quaternion.hpp]: maybe here you can add a typedef for Eigen::QuaternionBase<D> as Base

File: src/math/quaternion.hpp [hunk 3]
```diff
@@ -66,19 +66,32 @@ namespace se3
...
-    const Scalar alpha = ((Scalar)3 - q.squaredNorm()) / 2;
+    const Scalar N2 = q.squaredNorm();
+#ifndef NDEBUG
+    const Scalar epsilon = sqrt(sqrt(Eigen::NumTraits<Scalar>::epsilon()));
+    assert(std::fabs(N2-1.) <= epsilon);
+#endif
+    const Scalar alpha = ((Scalar)3 - N2) / 2;
     const_cast <Eigen::QuaternionBase<D> &> (q).coeffs() *= alpha;
+#ifndef NDEBUG
+    const Scalar M = 3 * std::pow((Scalar)1-epsilon, (Scalar)-5/2) / 4;
```
Comment (jcarpent) [inline src/math/quaternion.hpp]: 5/2 might be interpreted as a integer division

**Comment by jcarpent** [comment]
Seems good to me. Wait for travis.

**Pull request merged**